Vector.hpp

```diff
@@ -0,0 +1,246 @@
+#ifndef VECTOR_HPP
+
+#define VECTOR_HPP
+
+#include <iostream>
+
+
+template<typename T>
+class Vector
+{
+public:
+	/**
+	 * @brief Default constructor.
+	 *
+	 * Constructs an empty Vector with an initial capacity of 2.
+	 */
+	Vector()
+	{
+		Realloc(2);
```

Comment:
that's a magic number. 
use constats. make sure the constants are calculated during compile time and not during runtime